### PR TITLE
fixes #6428 correct actual duration time in statistics

### LINKED ISSUE #6428: Statistics : real ticket duration time is wrong
**Describe the bug**

Real ticket duration time calculation is wrong in statistics view.

**Page(s) URL**

/front/stat.tracking.php?itemtype=Ticket&type=technicien&_date1=08-10-2018&date1=2018-10-08&submit=Display+report&_date2=08-10-2019&date2=2019-10-08&value2=0&showgraph=0

**To reproduce**

0. Make sure, you don't have any ticket resolved today
1. Create one first ticket («A») assigned to a technician
2. Create a «done» task for this ticket with a duration of 10 min.
3. Resolve this ticket
4. Create a second ticket («B») assigned to the same technician
5. Resolve this ticket (without task)
6. In statistics view «Tickets - by ticket», choose «Assigned to - Technician as assigned» and the proper parameters to see only the two tickets you've just created
7. «Total duration» under «Real duration» is 20 min.

**Expected behavior**

The total duration should be 10 min.
Without any task in ticket «B», we should consider there is no duration for this ticket.

### PR DESCRIPTION
**Pull request by grosmanal**
Get the number of resolved tickets with action time in order to compute
correct total action time

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | #6428

### CODE REVIEW

File: inc/stat.class.php [hunk 1]
```diff
@@ -635,20 +635,26 @@ static function showTable($itemtype, $type, $date1, $date2, $start, array $value
             }
             echo Search::showItem($output_type, $timedisplay, $item_num, $row_num);
 
+            //le nombre d'interventions résolues ayant une durée
```
Comment (trasher) [inline inc/stat.class.php]: only in english please ;)

Comment (grosmanal) [reply inc/stat.class.php]: Ok. I just copy / paste code with french comments

File: inc/stat.class.php [hunk 2]
```diff
@@ -1063,6 +1068,28 @@ static function constructEntryValues($itemtype, $type, $begin = "", $end = "", $
...
             break;
 
+         case "inter_solved_with_actiontime" :
+            $WHERE["$table.status"] = $solved_status;
+            $WHERE["$table.actiontime"] = ['>', 0];
+            $WHERE[] = ['NOT' => ["$table.solvedate" => null]];
+            $WHERE[] = getDateCriteria("$table.solvedate", $begin, $end);
+
+            $date_unix = new QueryExpression(
+               "FROM_UNIXTIME(UNIX_TIMESTAMP(`$table`.`solvedate`),'%Y-%m') AS date_unix"
```
Comment (trasher) [inline inc/stat.class.php]: Please avoid hardcode backquotes, use `$DB->quoteName('$table.$solvedate')`

File: inc/stat.class.php [hunk 3]
```diff
@@ -1063,6 +1068,28 @@ static function constructEntryValues($itemtype, $type, $begin = "", $end = "", $
...
             break;
 
+         case "inter_solved_with_actiontime" :
+            $WHERE["$table.status"] = $solved_status;
+            $WHERE["$table.actiontime"] = ['>', 0];
+            $WHERE[] = ['NOT' => ["$table.solvedate" => null]];
+            $WHERE[] = getDateCriteria("$table.solvedate", $begin, $end);
+
+            $date_unix = new QueryExpression(
+               "FROM_UNIXTIME(UNIX_TIMESTAMP(" . $DB->quoteName("$table.solvedate") . "),'%Y-%m') AS date_unix"
```
Comment (cedric-anne) [inline inc/stat.class.php]:
```suggestion
               "FROM_UNIXTIME(UNIX_TIMESTAMP(" . $DB->quoteName("$table.solvedate") . "),'%Y-%m') AS ".$DB->quoteName('date_unix')
```

**Pull request merged**